org.opent2t.sample.binaryswitch.superpopular/com.contosothings.binaryswitch/js/thingTranslator.js

```diff
@@ -1,5 +1,14 @@
 'use strict';
 var OpenT2TError = require('opent2t').OpenT2TError;
+var crypto = require('crypto');
+
+/**
+ * Generate a GUID for a given ID.
+ */
+function generateGUID(stringID) {
+    var guid = crypto.createHash('sha1').update('Contoso' + stringID).digest('hex');
+    return guid.substr(0, 8) + '-' + guid.substr(8, 4) + '-' + guid.substr(12, 4) + '-' + guid.substr(16, 4) + '-' + guid.substr(20, 12);
```

Comment:
like "replace(/([0-z]{8})([0-z]{4})([0-z]{4})([0-z]{4})([0-z]{12})/,"$1-$2-$3-$4-$5")"?

---
In reply to: [108277675](https://github.com/openT2T/translators/pull/217#discussion_r108277675) [](ancestors = 108277675)